packages/suite/src/views/wallet/coinmarket/exchange/index.tsx

```diff
@@ -1,8 +1,47 @@
 import React from 'react';
-import { CoinmarketLayout } from '@wallet-components';
+import { CoinmarketLayout, WalletLayout } from '@wallet-components';
+import { AppState } from '@suite-types';
+import { ComponentProps, Props } from '@wallet-types/coinmarketExchangeForm';
+import { connect } from 'react-redux';
+import ExchangeForm from './components/ExchangeForm';
+import {
+    useCoinmarketExchangeForm,
+    ExchangeFormContext,
+} from '@wallet-hooks/useCoinmarketExchangeForm';
 
-const CoinmarketBuy = () => {
-    return <CoinmarketLayout>Coming soon</CoinmarketLayout>;
+const mapStateToProps = (state: AppState): ComponentProps => ({
+    selectedAccount: state.wallet.selectedAccount,
+    quotesRequest: state.wallet.coinmarket.exchange.quotesRequest,
+    exchangeCoinInfo: state.wallet.coinmarket.exchange.exchangeCoinInfo,
+    fiat: state.wallet.fiat,
+    device: state.suite.device,
+    localCurrency: state.wallet.settings.localCurrency,
+    fees: state.wallet.fees,
+});
+
+const CoinmarketExchangeLoaded = (props: Props) => {
+    const { selectedAccount } = props;
+    const coinmarketExchangeContextValues = useCoinmarketExchangeForm({
+        ...props,
+        selectedAccount,
+    });
+
+    return (
+        <CoinmarketLayout>
+            <ExchangeFormContext.Provider value={coinmarketExchangeContextValues}>
+                <ExchangeForm />
+            </ExchangeFormContext.Provider>
+        </CoinmarketLayout>
+    );
+};
+
+const CoinmarketExchange = (props: ComponentProps) => {
+    const { selectedAccount } = props;
+    if (selectedAccount.status !== 'loaded') {
+        return <WalletLayout title="TR_NAV_EXCHANGE" account={selectedAccount} />;
+    }
+
+    return <CoinmarketExchangeLoaded {...props} selectedAccount={selectedAccount} />;
 };
 
-export default CoinmarketBuy;
+export default connect(mapStateToProps)(CoinmarketExchange);
```

Comment:
lets use hooks instead of connected component from now on, like it was [changed here](https://github.com/trezor/trezor-suite/commit/2fa02c3e3711c6c9c1cb9019d784eb4cf5d86942)

you don't have to introduce children props, but it would be useful to write hooks tests

Reply:
Done, for now without children